ql/src/java/org/apache/hadoop/hive/ql/optimizer/SortedDynPartitionOptimizer.java

```diff
@@ -409,26 +409,54 @@ private boolean removeRSInsertedByEnforceBucketing(FileSinkOperator fsOp) {
       // and grand child
       if (found) {
         Operator<? extends OperatorDesc> rsParent = rsToRemove.getParentOperators().get(0);
-        Operator<? extends OperatorDesc> rsChild = rsToRemove.getChildOperators().get(0);
-        Operator<? extends OperatorDesc> rsGrandChild = rsChild.getChildOperators().get(0);
-
-        if (rsChild instanceof SelectOperator) {
-          // if schema size cannot be matched, then it could be because of constant folding
-          // converting partition column expression to constant expression. The constant
-          // expression will then get pruned by column pruner since it will not reference to
-          // any columns.
-          if (rsParent.getSchema().getSignature().size() !=
-              rsChild.getSchema().getSignature().size()) {
+        List<Operator<? extends OperatorDesc>> rsChildren = rsToRemove.getChildOperators();
+
+        Operator<? extends OperatorDesc> rsChildToRemove = null;
+
+        for (Operator<? extends OperatorDesc> rsChild : rsChildren) {
```

Comment:
In which case would we have a RS with multiple children? Can we leave a comment explaining it? Otherwise, we should add a Precondition with number of children 1.

Reply:
I assumed that this could be possibility and therefore accounted for it, but if this assumption is wrong I update the code.

Reply:
Yes, there should not be a RS with multiple children, we can simplify that code. You can even add an assert to the new code to make sure.

Reply:
@jcamachor I have addressed in latest comment.

Reply:
@jcamachor all tests passed.